### PR TITLE
Add support for blend modes, image opacity and radial gradient

### PR DESCRIPTION
**Pull request by bodymovin**
PR to add support for blend modes, radial gradients and opacity for image elements.

### CODE REVIEW

File: src/svg/SkSVGDevice.cpp [hunk 1]
```diff
@@ -655,6 +662,48 @@ SkString SkSVGDevice::AutoElement::addLinearGradientDef(const SkShaderBase::Grad
...
+                                                        const SkMatrix& localMatrix) {
+    SkASSERT(fResourceBucket);
+    SkString id = fResourceBucket->addLinearGradient();
+
+    {
+        AutoElement gradient("radialGradient", fWriter);
+
+        gradient.addAttribute("id", id);
+        gradient.addAttribute("gradientUnits", "userSpaceOnUse");
+        gradient.addAttribute("cx", info.fPoint[0].x());
```
Comment (bodymovin) [inline src/svg/SkSVGDevice.cpp:676]: at Rive we don't support an end circle position, so cx and cy need to point to the start circle as well.

File: src/svg/SkSVGDevice.cpp [hunk 2]
```diff
@@ -960,6 +1009,61 @@ void SkSVGDevice::drawPath(const SkPath& path, const SkPaint& paint, bool pathIs
...
+            case SkBlendMode::kColor:
+            blendMode = "color";
+            break;
+            case SkBlendMode::kLuminosity:
+            blendMode = "luminosity";
+            break;
+        }
+        if(!blendMode.empty()) {
+            std::string styleValue = "mix-blend-mode: " + blendMode;
+            elem.addAttribute("style", styleValue.c_str());
```
Comment (bodymovin) [inline src/svg/SkSVGDevice.cpp:1064]: SVGs don't support blend modes as [presentation attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/Presentation), they only work as a style properties.
Figma exports svg blend modes this way, but illustrator exports them as presentation attributes, which fails (even when reimporting them in illustrator itself!).
So I'm using the "style" approach, which works when opening them in the browser and importing them in figma, but it gets lost when importing them in illustrator.

File: src/svg/SkSVGDevice.cpp [hunk 3]
```diff
@@ -655,6 +662,48 @@ SkString SkSVGDevice::AutoElement::addLinearGradientDef(const SkShaderBase::Grad
...
+        gradient.addAttribute("gradientUnits", "userSpaceOnUse");
+        gradient.addAttribute("cx", info.fPoint[0].x());
+        gradient.addAttribute("cy", info.fPoint[0].y());
+        gradient.addAttribute("fx", info.fPoint[0].x());
+        gradient.addAttribute("fy", info.fPoint[0].y());
+        gradient.addAttribute("r", info.fRadius[0]);
+        gradient.addAttribute("fr", info.fRadius[1]);
+
+        if (!localMatrix.isIdentity()) {
+            this->addAttribute("gradientTransform", svg_transform(localMatrix));
```
Comment (csmartdalton) [inline src/svg/SkSVGDevice.cpp:684]: This is great as-is, but FYI I don't think we ever use the localMatrix

File: src/svg/SkSVGDevice.cpp (hunk 2 above)
Comment (csmartdalton) [reply src/svg/SkSVGDevice.cpp:1064]: What is `mix-blend-mode`? Is this an SVG thing or did we define it somewhere?

**Review by csmartdalton — APPROVED** [review]
Nice! This ended up being super concise. I added a couple questions for my own understanding but this LGTM!

File: src/svg/SkSVGDevice.cpp (hunk 2 above)
Comment (bodymovin) [reply src/svg/SkSVGDevice.cpp:1064]: it's a css/svg thing. That's how blend modes are defined in HTML and SVG
https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode

**Pull request merged**